### PR TITLE
refresh serverbrowser when disconnecting / abort

### PR DESCRIPTION
**Pull request by luk51**
This might be not complete, but this adds the functionality to refresh the server browser list after disconnecting from a server, or after aborting joining a server.

This fixes issue #5863.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus.cpp [hunk 1]
```diff
@@ -1783,6 +1783,24 @@ int CMenus::Render()
 			{
 				Client()->Disconnect();
 				m_Popup = POPUP_NONE;
+				if(g_Config.m_UiPage == PAGE_INTERNET)
```
Comment (def-) [inline src/game/client/components/menus.cpp:1786]: indentation

Comment (luk51) [reply src/game/client/components/menus.cpp:1786]: ran fix_style.py

File: src/game/client/components/menus_ingame.cpp [hunk 2]
```diff
@@ -61,6 +61,24 @@ void CMenus::RenderGame(CUIRect MainView)
 		else
 		{
 			Client()->Disconnect();
+			if(g_Config.m_UiPage == PAGE_INTERNET)
```
Comment (def-) [inline src/game/client/components/menus_ingame.cpp:64]: I feel like we have this code in more than one place already, now 2 more. Put it into a helper method.

Comment (luk51) [reply src/game/client/components/menus_ingame.cpp:64]: Ok. Any other helper method that is similar structured, so I can look how it is done?

Comment (def-) [reply src/game/client/components/menus_ingame.cpp:64]: CMenus::SetMenuPage

Comment (luk51) [reply src/game/client/components/menus_ingame.cpp:64]: ok, but it sometimes does this:
```
else if(g_Config.m_UiPage == PAGE_DDNET)
			ServerBrowser()->Refresh(IServerBrowser::TYPE_DDNET);
			```
and sometimes does more
```
else if(g_Config.m_UiPage == PAGE_DDNET)
		{
			// start a new server list request
			Client()->RequestDDNetInfo();
			ServerBrowser()->Refresh(IServerBrowser::TYPE_DDNET);
		}
```
it only does that call: `Client()->RequestDDNetInfo();` for ddnet and kog tab.
why, and how to 'summarize' this in the helper function?

Comment (luk51) [reply src/game/client/components/menus_ingame.cpp]: @def- ok, i tried my best. but now the helper function calls `Client()->RequestDDNetInfo();`, so at some points the call is made, where it was not before. I don't know if that's bad / wanted or not. Maybe it's possible to exclude the call from the helper function?

**Comment by def-** [comment]
I think that's fine. Thanks for the change.
bors r+

**Review by def- — APPROVED** [review]

**Pull request merged**